dateutil/test/test_rrule.py

```diff
@@ -4912,3 +4912,17 @@ def testWeekdayReprWithN(self):
 
         for repstr, wday in zip(with_n_reprs, with_n_wdays):
             self.assertEqual(repr(wday), repstr)
+
+    def testMalformedRuleErrorWhenTimeFormatIsNotCorrect(self):
+        rule = "FREQ=YEARLY;WKST=MO;UNTIL=2019-03-29T00:59:59+01:00"
+        with pytest.raises(ValueError) as exception:
+            rrulestr(rule, dtstart=datetime(1997, 9, 2, 9, 0))
+        self.assertEqual("Malformed rule: " + rule,
+                         str(exception.value))
```

Comment:
Also, if we replace the catch-all error message, we probably should not add any tests for the message content either, but just check for a `ValueError`.